### PR TITLE
improve naming to represent intent; use `std::generate`

### PR DESCRIPTION
**Pull request by xtofl**

### CODE REVIEW

File: examples/bounding_box.cu [hunk 1]
```diff
@@ -66,21 +66,17 @@ int main(void)
...
   // generate some random points in the unit square
-  for(size_t i = 0; i < N; i++)
-  {
-      float x = u01(rng);
-      float y = u01(rng);
-      points[i] = point2d(x,y);
-  }
+  std::generate(points.begin(), points.end(), [&] {
+    auto x = u01(rng);
+    auto y = u01(rng);
```
Comment (brycelelbach) [inline examples/bounding_box.cu]: Let's avoid auto here, we're not requiring C++11 yet.

Comment (xtofl) [reply examples/bounding_box.cu]: If we're not requiring C++11, we can't use this lambda either, can we?

**Comment by brycelelbach** [comment]
This was closed when we pushed the new Git master history to GitHub. Please rebase onto the new master and we'll take a look.

**Comment by xtofl** [comment]
That was easy.  Rebased/pushed.

**Comment by xtofl** [comment]
Is something happening to this pull request?  Should I close it?

**Comment by brycelelbach** [comment]
Hey, we will take a look at it, but we are a little backed up with our release cycle right now

Sent from Nine<http://www.9folders.com/>
________________________________

**Comment by xtofl** [comment]
No problem.  It is, after all, only a tiny first step in pouring 'modern C++' over the thrust examples.

I think it's part of a bigger effort to take on the whole example code.  To me, it's important that examples breath good style, so that when I point people at them, they fullful a larger educational purpose than just showcasing the library.  I'll make an Issue (#1041) out of that, as to formalize it.

**Pull request closed without merging**